src/main/scala/edu/cmu/cs/obsidian/parser/AST.scala

```diff
@@ -64,95 +89,182 @@ sealed abstract class InvokableDeclaration() extends Declaration {
+
+case class Parent() extends AtomicExpression //todo type of parent?
+
+case class Conjunction(e1: Expression, e2: Expression) extends BinaryExpression(Conjunction, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(BoolType())
+}
+
+case class Disjunction(e1: Expression, e2: Expression) extends BinaryExpression(Disjunction, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(BoolType())
+}
+
+case class LogicalNegation(e: Expression) extends UnaryExpression(LogicalNegation, e) {
+    override val obstype: Option[ObsidianType] = Some(BoolType())
+}
+
+
+case class Add(e1: Expression, e2: Expression) extends BinaryExpression(Add, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(IntType())
+}
+
+case class StringConcat(e1: Expression, e2: Expression) extends BinaryExpression(StringConcat, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(StringType())
+}
+
+case class Subtract(e1: Expression, e2: Expression) extends BinaryExpression(Subtract, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(IntType())
+}
+
+case class Divide(e1: Expression, e2: Expression) extends BinaryExpression(Divide, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(IntType())
+}
+
+case class Multiply(e1: Expression, e2: Expression) extends BinaryExpression(Multiply, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(IntType())
+}
+
+case class Mod(e1: Expression, e2: Expression) extends BinaryExpression(Mod, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(IntType())
+}
+
+case class Negate(e: Expression) extends UnaryExpression(Negate, e) {
+    override val obstype: Option[ObsidianType] = Some(IntType())
+}
+
+case class Equals(e1: Expression, e2: Expression) extends BinaryExpression(Equals, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(BoolType())
+}
+
+case class GreaterThan(e1: Expression, e2: Expression) extends BinaryExpression(GreaterThan, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(BoolType())
+}
+
+case class GreaterThanOrEquals(e1: Expression, e2: Expression) extends BinaryExpression(GreaterThanOrEquals, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(BoolType())
+}
+
+case class LessThan(e1: Expression, e2: Expression) extends BinaryExpression(LessThan, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(BoolType())
+}
+
+case class LessThanOrEquals(e1: Expression, e2: Expression) extends BinaryExpression(LessThanOrEquals, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(BoolType())
+}
+
+case class NotEquals(e1: Expression, e2: Expression) extends BinaryExpression(NotEquals, e1, e2) {
+    override val obstype: Option[ObsidianType] = Some(BoolType())
+}
+
 case class Dereference(e: Expression, f: String) extends UnaryExpression(Dereference(_, f), e) {
+    override val obstype: Option[ObsidianType] = e.obstype // todo
```

Comment:
Why todo? Since obstype is immutable, I think this is safe.

Reply:
OK! I marked it as a todo because I was kind of guessing, but if it "smells right" to you then I'll go with it.